src/git/config.go

```diff
@@ -301,27 +277,33 @@ func getLocalConfigurationValueWithDefault(key, defaultValue string) string {
 func getConfigurationKeysMatching(toMatch string) (result []string) {
 	configRegexp, err := regexp.Compile(toMatch)
 	exit.IfWrapf(err, "Error compiling configuration regular expression (%s): %v", toMatch, err)
-	lines := command.New("git", "config", "-l", "--local", "--name").Output()
-	for _, line := range strings.Split(lines, "\n") {
-		if configRegexp.MatchString(line) {
-			result = append(result, line)
+	for key := range configMap.Data() {
+		if configRegexp.MatchString(key) {
+			result = append(result, key)
```

Comment:
ConfigMap already provides `Get` to read simple values. This functionality of finding all keys that match a given regex could also be part of that class, since it is also some form of more complex "getter":

```go
func (c *ConfigMap) KeyMatching(re *Regexp) (result map[string]) {
  for key := range c.Data() {
    if configRegexp.MatchString(key) {
      result = append(result, key)
    }
  }
  return
}
```

This has the advantage that ConfigMap doesn't have to expose its internal data structure via the `Data` function, which creates more coupling between this class and ConfigMap's internals.